resources/build/builder.inc.sh

```diff
@@ -0,0 +1,680 @@
+    # None of the answers are 100% correct for cross-platform
+    # On macOS, requires coreutils (`brew install coreutils`)
+    local SCRIPT=$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")
+    REPO_ROOT=$(dirname $(dirname $(dirname "$SCRIPT")))
+    readonly REPO_ROOT
+}
+
+_builder_findRepoRoot
+
+# Used to build script-related build variables useful for referencing the calling script
+# and for prefixing builder_finish_action outputs in order to more clearly identify the calling
+# script.
+#
+# Assumes that `findKeymanRoot` has already been called, a condition met later on
+# within this script.
+function _builder_setBuildScriptIdentifiers() {
+  if [ ! -z ${THIS_SCRIPT+x} ]; then
+    THIS_SCRIPT_PATH="$(dirname "$THIS_SCRIPT")"
+    readonly THIS_SCRIPT_PATH
+    THIS_SCRIPT_NAME="$(basename "$THIS_SCRIPT")"
+    readonly THIS_SCRIPT_NAME
+    # Leaves only the part of the path based upon KEYMAN_ROOT.
+    THIS_SCRIPT_IDENTIFIER=${THIS_SCRIPT_PATH#"$KEYMAN_ROOT/"}
+    readonly THIS_SCRIPT_IDENTIFIER
+  fi
+}
+
+################################################################################
+# Standard build script functions for managing command line, actions and targets
+################################################################################
+
+# The following allows coloring of warning and error lines, but only works if there's a
+# terminal attached, so not on the build machine.
+
+# Overrides default colorization of logging; can be used in command-line with
+# --color or --no-color, or overridden as necessary on a per-script basis.
+#
+# Parameters
+#  1: use_color       true or false
+builder_use_color() {
+  if $1; then
+    COLOR_RED=$(tput setaf 1)
+    COLOR_GREEN=$(tput setaf 2)
+    COLOR_YELLOW=$(tput setaf 3)
+    COLOR_BLUE=$(tput setaf 4)
+    COLOR_PURPLE=$(tput setaf 5)
+    COLOR_TEAL=$(tput setaf 6)
+    COLOR_WHITE=$(tput setaf 7)
+    COLOR_GREY=$(tput setaf 8)
+    COLOR_RESET=$(tput sgr0)
+    # e.g. VSCode https://code.visualstudio.com/updates/v1_69#_setmark-sequence-support
+    HEADING_SETMARK='\x1b]1337;SetMark\x07'
+
+    # Used by `builder_display_usage` when marking special terms (actions, targets, options)
+    # in the plain-text description area.
+    BUILDER_TERM_START="$COLOR_BLUE"
+    BUILDER_TERM_END="$COLOR_RESET"
+  else
+    COLOR_RED=
+    COLOR_GREEN=
+    COLOR_YELLOW=
+    COLOR_BLUE=
+    COLOR_PURPLE=
+    COLOR_TEAL=
+    COLOR_WHITE=
+    COLOR_GREY=
+    COLOR_RESET=
+    HEADING_SETMARK=
+    BUILDER_TERM_START="<"
+    BUILDER_TERM_END=">"
+  fi
+}
+
+if [[ -n "$TERM" ]] && [[ "$TERM" != "dumb" ]] && [[ "$TERM" != "unknown" ]]; then
+  builder_use_color true
+else
+  builder_use_color false
+fi
+
+function die () {
+    # TODO: consolidate this with fail() from shellHelperFunctions.sh
+    echo
+    echo "${COLOR_RED}$* ${COLOR_RESET}"
+    echo
+    exit 1
+}
```

Comment:
Leave `die()` in build-utils.sh, but make `builder_die()` do the work, i.e. swap the definitions. `builder_die()` stays in builder.inc.sh of course.